File: mobile_app/lib/core/models/vehicle.dart
class Vehicle {
  final int? id;
  final String nickname;
  final String licensePlate;
  final String vehicleType; // 'auto' | 'busje'
  final bool isCurrent;

  const Vehicle({
    this.id,
    required this.nickname,
    required this.licensePlate,
    required this.vehicleType,
    this.isCurrent = false,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) => Vehicle(
    id: json['id'] as int?,
    nickname: json['nickname'] as String,
    licensePlate: json['license_plate'] as String,
    vehicleType: json['vehicle_type'] as String,
    isCurrent: json['is_current'] as bool? ?? false,
  );

  Map<String, dynamic> toJson() => {
    if (id != null) 'id': id,
    'nickname': nickname,
    'license_plate': licensePlate,
    'vehicle_type': vehicleType,
    'is_current': isCurrent,
  };
}

File: mobile_api/serializers/vehicle.py
import re
from rest_framework import serializers
from accounts.models import VehicleType


def normalize_license_plate(value: str) -> str:
    return re.sub(r"[^A-Z0-9]", "", value.upper().strip())


class MobileVehicleSerializer(serializers.Serializer):
    id = serializers.IntegerField(read_only=True)
    license_plate = serializers.CharField(max_length=20)
    nickname = serializers.CharField(max_length=60, allow_blank=True, required=False)
    vehicle_type = serializers.ChoiceField(
        choices=[VehicleType.AUTO, VehicleType.BUSJE],
        default=VehicleType.AUTO,
    )
    is_current = serializers.BooleanField(default=False)

    def validate_license_plate(self, value):
        normalized = normalize_license_plate(value)

        if len(normalized) < 5 or len(normalized) > 8:
            raise serializers.ValidationError("Vul een geldig kenteken in.")

        return normalized
